slot-all.html

```diff
@@ -0,0 +1,140 @@
+                }
+                // Create slots for initial list of children
+                this._createSlots(Array.from(host.children));
+            }
+            /**
+             * Disconnect host observer, remove created slots, clear the slots map.
+             */
+            disconnectedCallback(){
+                // unobserve host changes
+                this._hostObserver && this._hostObserver.disconnect();
+                // unobserve shady dom host changes
+                // workaround https://github.com/webcomponents/shadydom/issues/260
+                this.__shadyObserver && ShadyDOM.unobserveChildren(this.__shadyObserver);
+                for(let [name, slot] of this.__createdSlots){
+                    // workaround https://github.com/webcomponents/shadydom/issues/262
+                    // WC polyfill does not support .remove()
+                    // slot.remove();
+                    slot.parentNode && slot.parentNode.removeChild(slot);
+                }
+                this.__createdSlots.clear();
+            }
+            /**
+             * Create slots for given elements, if needed:
+             *  - element has a slot name,
+             *  - names slot was not yet created by this `slot-all``
+             * @param  {NodeList|HTMLCollection} nodeList list of host's children
+             */
+            _createSlots(nodeList){
+                const slotAll = this;
+                const slotsMap = slotAll.__createdSlots;
+
+                nodeList.forEach((element)=>{
+                    const name = element.slot;
+                    if(
+                        element.nodeType === Node.ELEMENT_NODE &&
+                        element.hasAttribute('slot') &&
+                        !slotsMap.has(name)
+                        // we do create a slot for elements already
+                        // assigned by other `slot(-all)`s
+                        // to handle distribution when others are removed.
+                        // !element.assignedSlot
+                    ){
+                        const namedSlot = document.createElement('slot');
+                        namedSlot.setAttribute('name', name);
+                        namedSlot.setAttribute('all-slotted', '');
+                        // add to the local map, to be able to clean it later
+                        slotsMap.set(name, namedSlot);
+                        // following is not supported by polyfill
+                        // slotAll.insertAdjacentElement('beforebegin', namedSlot);
+                        // workaround https://github.com/webcomponents/shadydom/issues/263
+                        slotAll.parentNode.insertBefore(namedSlot, slotAll);
+                    }
+                });
+            }
+            /**
+             * Remove slots for given elements, if needed:
+             *  - there is no more host's children with a slot name for given slot name
```

Comment:
`there is no more host's children with a slot name for given slot name`

↓

`there are no more host's children with a slot name for the given slot name`